### PR TITLE
Enable IDE0200 (remove unnecessary lambda expression)

### PR DESCRIPTION
**Pull request by elachlan**
Enables IDE0200 (remove unnecessary lambda expression).

Related #7887

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7916)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.cs [hunk 1]
```diff
@@ -218,7 +218,7 @@ public override object EditValue(ITypeDescriptorContext context, IServiceProvide
             _currentContext = context;
 
             // Child modal dialog - launching in SystemAware mode.
-            CollectionForm localCollectionForm = DpiHelper.CreateInstanceInSystemAwareContext(() => CreateCollectionForm());
+            CollectionForm localCollectionForm = DpiHelper.CreateInstanceInSystemAwareContext(CreateCollectionForm);
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.cs:221]: IIRC, this used to have inferior perf characteristics, I have to dig for details though...

Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.cs:221]: This is no longer the case based on https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7/#analyzers

I think this was the related change https://github.com/dotnet/roslyn/pull/58288

**Review by dreddy-work — APPROVED** [review]
LGTM

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CollectionEditor.cs:221]: Thank you for the link, this is what I was referring to.

**Pull request merged**